install_files/ansible-base/roles/restore/tasks/main.yml

```diff
@@ -1,22 +1,81 @@
 ---
-- name: Copy restore script to Application Server.
-  copy:
-    src: restore.py
-    dest: /tmp/
-    owner: root
-    mode: "0770"
-
-- name: Copy backup tarball of config info to Application Server.
-  copy:
+- name: Create temporary directory for Tor configuration check
+  connection: local
+  become: no
+  tempfile:
+    state: directory
+  register: torrc_check_dir
+
+- name: Fetch current Tor configuration from app server
+  become: no
+  fetch:
+    src: /etc/tor/torrc
+    dest: "{{ torrc_check_dir.path }}"
+
+- name: Create directory to hold the Tor configuration from the backup
+  connection: local
+  become: no
+  file:
+    path: "{{ torrc_check_dir.path }}/backup"
+    state: directory
+
+- name: Extract Tor configuration from backup
+  connection: local
+  become: no
+  unarchive:
+    dest: "{{ torrc_check_dir.path }}/backup/"
     src: "{{ restore_file }}"
-    dest: /tmp/{{ restore_file }}
+    extra_opts:
+      - "etc/tor/torrc"
 
-- name: Run the restore script (can be slow).
-  command: /tmp/restore.py /tmp/{{ restore_file }}
-  async: 3600
-  poll: 10
+- name: Check for Tor configuration differences between the backup and server
+  connection: local
+  become: no
+  command: "python {{ role_path }}/files/compare_torrc.py {{ torrc_check_dir.path }}"
 
-- name: Delete the restore file to save disk space.
+- name: Remove temporary directory for Tor configuration check
+  connection: local
+  become: no
   file:
-    path: /tmp/{{ restore_file }}
+    path: "{{ torrc_check_dir.path }}"
     state: absent
+  when: torrc_check_dir.path is defined
+
+- name: Copy backup to application server
+  synchronize:
```

Comment:
Nit: `copy` module is sufficient here, since we're just copying a single file.

Reply:
The idea was that with `partial: yes` an interrupted upload of the backup could be resumed.